packages/discovery/src/discovery/analysis/TemplateService.ts

```diff
@@ -260,7 +265,7 @@ export class TemplateService {
       const sourcesHash = Hash256(hash)
       const matchingTemplates = this.findMatchingTemplatesByHash(
         sourcesHash,
-        contract.address,
+        ChainSpecificAddress.address(contract.address),
```

Comment:
Don't extract raw address. Pass ChainSpecificAddress and appropriately refactor `findMatchingTemplatesByHash`